From 7ff0530f7bc810556a42a34c31b58a3b247332ce Mon Sep 17 00:00:00 2001
From: Yu Qiaowei <cerf.yu@rock-chips.com>
Date: Thu, 20 Aug 2020 09:58:09 +0800
Subject: [PATCH] video/rockchip: rga2: Compatible with kernel module

rga2 will be compiled into a kernel object, because cmd_buf
is a global variable and the address is stored in the modules
section, which makes it unable to obtain the physical address
through virt_to_phy(), so the local variable cmd_reg is used.

Signed-off-by: Yu Qiaowei <cerf.yu@rock-chips.com>
Change-Id: I3c088e727365fa8d515238ad88b2edaaee2b9a2b
---
 drivers/video/rockchip/rga2/rga2_drv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/rga2/rga2_drv.c b/drivers/video/rockchip/rga2/rga2_drv.c
index 8c2caf75f9b9..b9b79ab1a0aa 100644
--- a/drivers/video/rockchip/rga2/rga2_drv.c
+++ b/drivers/video/rockchip/rga2/rga2_drv.c
@@ -922,14 +922,14 @@ static void rga2_try_set_reg(void)
 			rga2_copy_reg(reg, 0);
 			rga2_reg_from_wait_to_run(reg);
 
-			rga2_dma_flush_range(&rga2_service.cmd_buff[0], &rga2_service.cmd_buff[32]);
+			rga2_dma_flush_range(&reg->cmd_reg[0], &reg->cmd_reg[32]);
 
 			//rga2_soft_reset();
 
 			rga2_write(0x0, RGA2_SYS_CTRL);
 
 			/* CMD buff */
-			rga2_write(virt_to_phys(rga2_service.cmd_buff), RGA2_CMD_BASE);
+			rga2_write(virt_to_phys(reg->cmd_reg), RGA2_CMD_BASE);
 
 #if RGA2_DEBUGFS
 			if (RGA2_TEST_REG) {
-- 
2.35.3

